### ISSUE TITLE
Inconsistent forms

### ISSUE THREAD

**Issue by lmcnulty** [responsible-ai-collaborative/aiid#996]
Currently, changes sometimes have to be copied among the `SubmissionForm`, `IncidentReportForm`, and `IncidentForm`s. We may want to break them into smaller components that get reused.

**Comment by smcgregor** [comment]
@lmcnulty should this still be open?

**Comment by lmcnulty** [comment]
@smcgregor Yes. The different forms have diverged more since then, but I think they still have some logic that could be shared but isn't.

**Comment by smcgregor** [comment]
There is also the issue where "incident reports" are no longer the right naming for the form since reports can pertain to issues and will soon pertain to variants as well. I think @pdcp1 is likely to run into this one naturally in short order since he is going to be working on variants.
